src/MongoDB.Analyzer/Core/ExpressionAnalysis.cs

```diff
@@ -33,7 +33,8 @@ internal record ExpressionAnalysisNode(
     SyntaxNode OriginalExpression,
     string ArgumentTypeName,
     SyntaxNode RewrittenExpression,
-    ConstantsMapper ConstantsRemapper) :
+    ConstantsMapper ConstantsRemapper,
+    List<Location> Locations) :
```

Comment:
I won't be able to add Locations to Locations [] if Locations [] is immutable. Unless I create redundant ExpressionAnalysisContexts which defeats the entire purpose of this though

Reply:
Unless, if its ok, I can cache the Builders Expressions in BuildersAnalyzer.cs so that I don't have to generate MQL for the same stuff over and over again.

Reply:
Done